File: c/hash-table/happy-number.c
bool happy(int n, int *count)
{
    if(n == 1)
        return true;
    if(*count == 100 || n == 0)
        return false;

    int tmp = 0;
    int d;
    while(n > 0){
        d = n % 10;
        n /= 10;
        tmp += d * d;
    }

    *count++;
    return happy(tmp,count);
}

bool isHappy(int n) {
    int count = 0;
    return happy(n,&count);
}

File: python/hash-table/happy-number.py
class Solution(object):
    def __init__(self):
        self.count = 0

    def isHappy(self, n):
        if n == 1:
            return True
        if self.count == 100:
            return False

        d = 0
        tmp = 0
        while n > 0:
            d = n % 10
            n /= 10
            tmp += (d * d)

        self.count += 1
        return self.isHappy(tmp)
